cirq-core/cirq/sim/act_on_density_matrix_args.py

```diff
@@ -107,7 +108,63 @@ def copy(self) -> 'cirq.ActOnDensityMatrixArgs':
             axes=self.axes,
             qid_shape=self.qid_shape,
             prng=self.prng,
-            log_of_measurement_results=self.log_of_measurement_results.copy(),
+            log_of_measurement_results=self.log_of_measurement_results,
```

Comment:
Can't do this. The logs should be a copy when the state is copied. The idea is that you have a snapshot that you can go back to. If log reference is shared, then this is broken. The only reason this doesn't cause failures is because the iterator clears the log after each moment, so it circumstantially happens that this is always emptied between each use.